components/core/src/fs.rs

```diff
@@ -663,51 +678,6 @@ fn find_command_with_pathext(candidate: &PathBuf) -> Option<PathBuf> {
 /// See, for example, Linux capabilities.
 pub fn am_i_root() -> bool { *EUID == 0u32 }
 
-/// Returns a `PathBuf` which represents the filesystem root for Habitat.
-///
-/// **Note** with the current exception of behavior on Windows (see below), an absolute default
-/// path of `"/"` should always be returned. This function is used to populate a one-time static
-/// value which cannot be altered for the execution length of a program. Packages in Habitat may
-/// contain binaries and libraries having dependent libraries which are located in absolute paths
-/// meaning that changing the value from this function will render existing packages un-runnable in
-/// the Supervisor. Furthermore as a rule in this codebase, external environment variables should
-/// *not* influence the behavior of inner libraries--any environment variables should be detected
-/// in a program at CLI parsing time and explicitly passed to inner module functions.
-///
-/// There is one exception to this rule which is supported for testing only--primarily exercising
-/// the Supervisor behavior. It allows setting a testing-only environment variable to influence the
-/// file system root for the duration of a running program.  Note that when using such an
-/// environment varible, any existing/actual Habitat packages may not run correctly due to the
-/// presence of absolute paths in package binaries and libraries. The environment variable will not
-/// be referenced, exported, or consumed anywhere else in the system to ensure that it is *only*
-/// used internally in test suites.
-///
```

Comment:
It would be good to preserve this knowledge in the updated code. Basically, FS_ROOT should only be changed via environment variable if you're doing things like limited testing or creating new self-contained root filesystems for things like tarballs or containers.

Reply:
I thought most of this sentiment was summed up in the comment on [`FS_ROOT_ENVVAR`](https://github.com/habitat-sh/habitat/blob/9df697676d01954ad9d214460e2c8b3289f933ee/components/core/src/fs.rs#L52). Do you think it is necessary to duplicate it on `FS_ROOT_PATH`?

Reply:
I liked this deleted comment because it goes more into _why_ it's important to not change it (rather than the more vague sense of "don't change it"). I think it would also be good to augment that with information about what situations you might legitimately need to change it.

It probably makes sense to have the comment be on `FS_ROOT_ENVVAR`, since that is the avenue by which you would be making such a change. Perhaps a "See FS_ROOT_ENVVAR documentation, also" note here would be nice, too.

Reply:
That makes sense, thanks! Addressed in (92592d)[https://github.com/habitat-sh/habitat/pull/7642/commits/92592d6b5b81b3ede057115635b383e9de9d259e).